libcudacxx/include/cuda/try_cancel

```diff
@@ -0,0 +1,324 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * NVIDIA SOFTWARE LICENSE
+ *
+ * This license is a legal agreement between you and NVIDIA Corporation ("NVIDIA") and governs your use of the
+ * NVIDIA/CUDA C++ Library software and materials provided hereunder (“SOFTWARE”).
```

Comment:
That license should be the simple libcu++ license

Reply:
I used what the `annotated_ptr` header was using.
Is there a better file I can use as the base for this one?